test/tensor_holder.hpp

```diff
@@ -216,7 +217,8 @@ struct tensor
                                     });
         seed ^= data.size();
         seed ^= desc.GetLengths().size();
-        std::srand(seed);
```

Comment:
#.1 tensor_holder change affects cache state.

Reply:
In that case I would suggest to have multiple independent generators, but it looks like it's too big change.

Reply:
done